packages/ra-core/src/form/FormWithRedirect.tsx

```diff
@@ -92,7 +93,38 @@ const FormWithRedirect: FC<FormWithRedirectProps> = ({
         [save]
     );
 
-    const formContextValue = useMemo(() => ({ setOnSave }), [setOnSave]);
+    const formContextValue = useMemo<FormContextValue>(
+        () => ({
+            setOnSave,
+            getGroupFields: name => formGroups.current[name] || [],
```

Comment:
As these functions are completely static, I suggest you define them outside of the component.